### DISCUSSION TITLE
I need help filtering a json field

### DISCUSSION THREAD

**Discussion by py-ficlub** [prisma/prisma#12997] [Q&A]
So I have a table with a json field, and that field has a nested field which is an array of objects. I'm trying to filter the rows by the nested array of objects to basically get rows where the nested array of object contains an object with a particular value. I keep getting 0 results. The database is PostgreSQL. Any help will be greatly appreciated. 

So the table this schema:
```
model Requests {
  ...,
 properties  Json?
}
```
The format of the saved properties is like this:
```

{
   multiple: true,
   receivers: [
   {
      name: '',
      id: 1,
      status: 'Pending'
    },
    {
      name: '',
      id: 5,
      status: 'Pending'
    },
   ]
}
```

And this is my query: 

```
const requests = await prisma.requests.findMany({
   where: {
        AND: [
            {
                'properties': {
                    path: ['receivers', '$[*].id'],
                    array_contains: 5,
                }
            },
            {
                'properties': {
                    path: ['receivers', '$[*].transactionStatus'],
                    array_contains: 'Pending'
                }
            }
        ]
    }
})
```

**Comment by Weakky** [comment]
Hey @py-ficlub,

There are a couple of things you need to know here.

First, filtering by the key values of objects inside an array is only supported by the MySQL connector [as documented here](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#path). Therefore the `$[*].transactionStatus` syntax is _not_ supported on Postgres.

Your best shot is to use `array_contains` and check whether a whole object is contained in the array. eg:

```ts
await prisma.testModel.findMany({
    where: {
      json: {
        path: ["receivers"],
        array_contains: [{"id": 5, "name": "", "status": "Pending"}],
      },
    },
  });
```

[As documented here](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#array_contains), the value of `array_contains` must always be an array.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
